src/scanpy/preprocessing/_simple.py

```diff
@@ -612,6 +612,47 @@ def normalize_per_cell(  # noqa: PLR0917
     return X if copy else None
 
 
+@numba.njit(cache=True, parallel=True)
+def to_dense(shape, indptr, indices, data):
```

Comment:
Is this really faster than `toarray`? In that case something like this should be contributed to scipy!

Also, please do types:

```suggestion
DT = TypeVar("DT")


def to_dense(shape: tuple[int, int], indptr: NDArray[np.integer], indices: NDArray[np.integer], data: NDArray[np.number]) -> NDArray[DT]:
```

Reply:
@flying-sheep , Yes it is faster then the toarray. for above example toarray was taking 8 sec vs numba kernel is taking 2 sec.

Reply:
wild. That should really go back into scipy as C++ or Cython code or whatever they use.